src/Features/Core/Portable/AddDebuggerDisplay/AbstractAddDebuggerDisplayCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,117 @@
+        }
+
+        private async Task<Document> ApplyAsync(Document document, TTypeDeclarationSyntax type, CancellationToken cancellationToken)
+        {
+            var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+            var generator = SyntaxGenerator.GetGenerator(document);
+
+            var newAttribute = generator
+                .Attribute("System.Diagnostics.DebuggerDisplayAttribute", generator.LiteralExpression("{ToString(),nq}"))
+                .WithAdditionalAnnotations(Simplifier.Annotation);
+
+            syntaxRoot = syntaxRoot.ReplaceNode(type, generator.AddAttributes(type, newAttribute).WithAdditionalAnnotations(s_trackingAnnotation));
+
+            syntaxRoot = await EnsureNamespaceImportAsync(
+                document,
+                generator,
+                syntaxRoot,
+                contextLocation: syntaxRoot.GetAnnotatedNodes(s_trackingAnnotation).Single(),
+                "System.Diagnostics",
+                cancellationToken).ConfigureAwait(false);
+
+            return document.WithSyntaxRoot(syntaxRoot);
+        }
+
+        private static async Task<SyntaxNode> EnsureNamespaceImportAsync(
+            Document document,
+            SyntaxGenerator generator,
+            SyntaxNode syntaxRoot,
+            SyntaxNode contextLocation,
+            string namespaceName,
+            CancellationToken cancellationToken)
+        {
+            var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+
+            var importsService = document.Project.LanguageServices.GetRequiredService<IAddImportsService>();
+            var newImport = generator.NamespaceImportDeclaration(namespaceName);
+
+            if (importsService.HasExistingImport(compilation, syntaxRoot, contextLocation, newImport))
+            {
+                return syntaxRoot;
+            }
+
+            var optionSet = await document.GetOptionsAsync(cancellationToken).ConfigureAwait(false);
+            var placeSystemNamespaceFirst = optionSet.GetOption(GenerationOptions.PlaceSystemNamespaceFirst, document.Project.Language);
+
+            return importsService.AddImport(
+                compilation,
+                syntaxRoot,
+                contextLocation,
+                newImport,
+                placeSystemNamespaceFirst);
+        }
+
+        protected abstract bool HasDebuggerDisplayAttribute(TTypeDeclarationSyntax typeDeclaration);
+
+        protected abstract bool IsToStringOverride(TMethodDeclarationSyntax methodDeclaration);
+
+        protected abstract bool DeclaresToStringOverride(TTypeDeclarationSyntax typeDeclaration);
+
+        protected bool IsDebuggerDisplayAttributeIdentifier(SyntaxToken name)
+        {
+            return name.ValueText == "DebuggerDisplay" || name.ValueText == "DebuggerDisplayAttribute";
+        }
```

Comment:
instead of doin things syntactically, just do it semantically, then you don't have to care about thsi.    That will also work for things like VB's case insensitivity.

Reply:
@CyrusNajmabadi I have tests that make sure this refactoring doesn't idiotically show up if the symbol isn't resolved for whatever reason (slow project loading, etc). Is that something that can also work 100% of the symbolic model when I get an error symbol back?

Reply:
> slow project loading

We shouldn't have refactorings running durin project load.  we fixed that since it was problematic for *all* of them.  if you're seeing that, it's a bug, and not something your particular feature has to care about.

> middle of a refactoring, etc). Is that something that can also work 100% of the symbolic model when I get an error symbol back?

Basically, we don't care about this elsewhere. So, absent an identified problem, just err toward just showing it, even if that means 1:1000 cases it is there unnecessarily.

We add conservative checks when something is showing up far too often in cases where it gets annoying.  I don't see how that could be the case here.